indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -604,13 +610,17 @@ private static DataSchema createDataSchema(
       @Nullable DimensionsSpec dimensionsSpec,
       @Nullable ClientCompactionTaskTransformSpec transformSpec,
       @Nullable AggregatorFactory[] metricsSpec,
-      @Nonnull ClientCompactionTaskGranularitySpec granularitySpec
+      @Nonnull ClientCompactionTaskGranularitySpec granularitySpec,
+      @Nullable CompactionRunner compactionRunner
   )
   {
     // Check index metadata & decide which values to propagate (i.e. carry over) for rollup & queryGranularity
     final ExistingSegmentAnalyzer existingSegmentAnalyzer = new ExistingSegmentAnalyzer(
         segments,
-        granularitySpec.isRollup() == null,
+        // For MSQ, always need rollup to check if there are some rollup segments already present.
+        compactionRunner == null || compactionRunner instanceof NativeCompactionRunner
```

Comment:
`compactionRunner` can never be `null`, it is already being initialized to a non-null value in the constructor.

Reply:
Removed the check